src/lib/y2network/dialogs/route.rb

```diff
@@ -0,0 +1,87 @@
+require "ipaddr"
+
+require "cwm/popup"
+require "y2network/route"
+require "y2network/widgets/destination"
+require "y2network/widgets/gateway"
+require "y2network/widgets/devices"
+require "y2network/widgets/gateway"
+require "y2network/widgets/route_options"
+
+Yast.import "Label"
+
+module Y2Network
+  module Dialogs
+    # Dialog to create or edit route.
+    class Route < CWM::Popup
+      # @param route is now term from table, but it should be object for single route
+      def initialize(route, available_devices)
+        log.info "route dialog with route: #{route.inspect} " \
+          "and devices #{available_devices.inspect}"
+        params = route.params
```

Comment:
Ok, this is a temporary solution (as `route` should be an object). But perhaps you could extract the elements to something meaningful:

```ruby
_, destination, gateway, _, interface, options = route.params
```

On the other hand, if it is going to change soon, maybe it does not worth.

Reply:
Well, my plan is probably object for route table, then can convert table Item to Route object and back. So ideally this dialog should in future receive simple Route object and responsibility for table conversion be on Table object ( not exist yet ).